crowbar_framework/app/models/pacemaker_service.rb

```diff
@@ -75,6 +75,151 @@ def create_proposal
     base
   end
 
+  # Small helper to expand all items (nodes, clusters) used inside an applied
+  # proposal
+  def expand_nodes_in_barclamp_role(cluster_role, node_object_all)
+    all_nodes_for_cluster_role = cluster_role.elements.values.flatten.compact.uniq
+
+    all_nodes_for_cluster_role_expanded, failures = expand_nodes_for_all(all_nodes_for_cluster_role)
+    unless failures.nil? || failures.empty?
+      @logger.debug "[pacemaker] expand_nodes_in_barclamp_role: skipping items that we failed to expand: #{failures.join(", ")}"
```

Comment:
This ```[pacemaker]``` should be replaced in future with the tagged logging capability of rails 4.